docs/manual/src/docs/asciidoc/_includes/reactive/oauth2/resource-server.adoc

```diff
@@ -1005,6 +1005,77 @@ ReactiveOpaqueTokenIntrospector introspector() {
 }
 ----
 
+[[oauth2resourceserver-multitenancy]]
+== Multi-tenancy
+
+A resource server is considered multi-tenant when there are multiple strategies for verifying a bearer token, keyed by some tenant identifier.
+
+For example, your resource server may accept bearer tokens from two different authorization servers.
+Or, your authorization server may represent a multiplicity of issuers.
+
+In each case, there are two things that need to be done and trade-offs associated with how you choose to do them:
+
+1. Resolve the tenant
+2. Propagate the tenant
+
+=== Resolving the Tenant By Claim
+
+One way to differentiate tenants is by the issuer claim. Since the issuer claim accompanies signed JWTs, this can be done with the `JwtIssuerReactiveAuthenticationManagerResolver`, like so:
+
+[source,java]
+----
+JwtIssuerReactiveAuthenticationManagerResolver authenticationManagerResolver = new JwtIssuerReactiveAuthenticationManagerResolver
+    ("https://idp.example.org/issuerOne", "https://idp.example.org/issuerTwo");
+
+http
+    .authorizeRequests(authorize -> authorize
+        .anyRequest().authenticated()
+    )
+    .oauth2ResourceServer(oauth2 -> oauth2
+        .authenticationManagerResolver(authenticationManagerResolver)
+    );
```

Comment:
Hello, would you mind update this document/example with a bit more details please?

I followed the guide, but the method `authenticationManagerResolver()` in [OAuth2ResourceServerConfigurer](https://github.com/spring-projects/spring-security/blob/master/config/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/resource/OAuth2ResourceServerConfigurer.java#L165) complained that it's expecting an `AuthenticationManagerResolver`, but was receiving `JwtIssuerReactiveAuthenticationManagerResolver`.

I tried to have my resource server may accept bearer tokens from two different authorization servers, using Spring Boot 2.3.1.RELEASE and spring-security-oauth2-resource-server 5.3.3.RELEASE

Sorry if I have missed something from the guide. Thanks

Reply:
Sorry to hear you are having trouble, @cmhuynh.

I believe the issue you've described is because you are trying to use a WebFlux class with the Web DSL. `AuthenticationManagerResolver` matches with `JwtIssuerAuthenticationManagerResolver` while `ReactiveAuthenticationManagerResolver` matches with `JwtIssuerReactiveAuthenticationManagerResolver`.

If you've got a concrete suggestion for how the docs can be improved to help with this issue, please open a separate ticket.